app.py

```diff
@@ -459,6 +467,20 @@ def delete_comment(id):
         flash("You are not authorized to delete this comment", "danger")
     return redirect(url_for("post", id=comment.post_id))
 
+@app.route("/posts/<id>/flag", methods=['POST', 'GET'])
+def report(id):
+    post = Post.query.filter_by(id=id).first()
+    ref = request.args.get('ref')
```

Comment:
https://www.programcreek.com/python/example/76929/flask.request.referrer has a quick helper to think about, but you can also redirect to the previous page. You're going to have a bad time if someone leaves off the `ref` for some reason.